src/components/Button/Button.module.css

```diff
@@ -229,45 +236,16 @@
   fill: currentColor; /* 1 */
 }
 
-/**
- * Button children before icon spacing
- * 1) Controls the spacing between the button children
- *    and the icon that comes _before_ button children
- */
-.button__icon + .button__children {
-  margin-left: var(--eds-size-1); /* 1 */
-
-  .button--sm & {
-    margin-left: var(--eds-size-half);
-  }
-}
-
-/**
- * Button after icon spacing
- * 1) Controls the spacing between the button children
- *    and the icon that comes _after_ button children
- */
-.button__children + .button__icon {
-  margin-left: var(--eds-size-1); /* 1 */
-
-  /**
-   * Button icon after button children within a link variant button
-   */
-  .button--link & {
-    margin-left: var(--eds-size-half);
-  }
-
-  .button--sm & {
-    margin-left: var(--eds-size-half);
-  }
-}
-
 /**
  * Loading icon rotiation style
  */
 .button.eds-is-loading > .button__icon {
```

Comment:
TODO: I should remove the `.button__icon` class name since these two are only being used together now.